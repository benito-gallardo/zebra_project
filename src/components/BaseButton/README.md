# BaseButton Component

A global button component with default styling.

## How to use

```
import React from 'react';

import BaseButton from  './components/BaseButton;

...
<Button link="/view-more" className="btn-class">View More</Button>

```

## Default Props

| Prop Name | Options                                                                     | Default Value | Description                                                                         |
| --------- | --------------------------------------------------------------------------- | ------------- | ----------------------------------------------------------------------------------- |
| link ,    | string                                                                      | `/`           | Sets the `href` attribute for the button.                                           |
| className | string                                                                      | `btn`         | Adds additional classes to the element. <br /> The default class is not overwriten. |
| children  | [React.Children API](https://reactjs.org/docs/react-api.html#reactchildren) | none          | Add inner content to the button: icon, text, ect.                                   |
