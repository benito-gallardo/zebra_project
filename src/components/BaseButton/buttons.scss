@import '../../assets/sass/vars/index';

.btn {
  display: inline-block;
  text-decoration: none;
  color: white;
  background: #5c6cff;
  padding: 0 1.5rem;
  font-size: 18px;
  font-weight: bold;
  border-radius: 50px;
  line-height: 50px;
  white-space: nowrap;
  .icon {
    margin-right: 0.5rem;
    &.btn-down-arrow {
      width: 18px;
    }
  }
  &.yellow-btn {
    background: $yellow-gradient;
  }
  &.blue-btn {
    background: $blue-gradient;
  }
}
