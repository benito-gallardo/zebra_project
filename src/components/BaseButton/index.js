import React from 'react';

import './buttons.scss';

const BaseButton = ({ link="/", className='', children }) => (
  <a href={link} className={`${className} btn`} data-testid="base-button">{children}</a>
)

export default BaseButton;