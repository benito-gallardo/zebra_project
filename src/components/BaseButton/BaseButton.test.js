import { render, cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';

import BaseButton from './index';

let Button;
afterEach(cleanup);

describe('The BaseButton Component ', () => { 

  it('renders without crashing', () => { 
    const div = document.createElement('div');
    ReactDOM.render(<BaseButton></BaseButton>, div);
  })

  it('matchs snapshot with props', () => { 
    const { asFragment } = render(<BaseButton link='/test' className='test-btn'><span>icon</span>View More</BaseButton>);
    expect(asFragment()).toMatchSnapshot();
  })

  it('matchs snapshot without props', () => { 
    const { asFragment } = render(<BaseButton></BaseButton>);
    expect(asFragment()).toMatchSnapshot();
  })

  describe('When props are provied ', () => { 
    beforeEach(() => {
      const { getByTestId } = render(<BaseButton link='/test' className='test-btn'><span>icon</span>View More</BaseButton>);
       Button = getByTestId('base-button');
    });
    
    it('when link prop is provied', () => { 
      expect(Button).toHaveAttribute('href', '/test'); 
    })

    it('when className prop is provided', () => { 
      expect(Button).toHaveClass('btn', 'test-btn');
    })

    it('when children are provied', () => { 
      expect(Button).toHaveTextContent('View More');
      expect(Button).toContainHTML('<span>icon</span>');
    })
  })

  describe('When props are not provied ', () => { 
    
    beforeEach(() => {
      const { getByTestId } = render(<BaseButton></BaseButton>);
      Button = getByTestId('base-button');
    });

    it('when link prop is not provied, default value should exist', () => { 
      expect(Button).toHaveAttribute('href', '/'); 
    })

    it('when className prop is not provided, default value should exist', () => { 
      expect(Button).toHaveClass('btn');
    })

    it('when children are not provied, children should\'t exist', () => { 
      expect(Button).toBeEmpty();
    })
  })
  
})
