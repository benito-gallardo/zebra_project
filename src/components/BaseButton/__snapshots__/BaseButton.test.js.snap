// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The BaseButton Component  matchs snapshot with props 1`] = `
<DocumentFragment>
  <a
    class="test-btn btn"
    data-testid="base-button"
    href="/test"
  >
    <span>
      icon
    </span>
    View More
  </a>
</DocumentFragment>
`;

exports[`The BaseButton Component  matchs snapshot without props 1`] = `
<DocumentFragment>
  <a
    class=" btn"
    data-testid="base-button"
    href="/"
  />
</DocumentFragment>
`;
